packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -0,0 +1,511 @@
+			appendLocalChangeToRevertibles(sharedString, interval, previousInterval, revertibles);
+		});
+		collection.change(id, 3, 8);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.removeIntervalById(id);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, []);
+		assertIntervals(sharedString2, collection2, []);
+	});
+	it("remote interval remove interacting with reverting an interval property change", () => {
+		sharedString.insertText(0, "hello world");
+		const id = collection.add(0, 5, IntervalType.SlideOnRemove, { foo: "one" }).getIntervalId();
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("propertyChanged", (interval, propertyDeltas, local, op) => {
+			appendLocalPropertyChangedToRevertibles(interval, propertyDeltas, revertibles);
+		});
+		collection.changeProperties(id, { foo: "two" });
+
+		collection2.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, []);
+		assertIntervals(sharedString2, collection2, []);
+	});
+	it("remote interval remove interacting with reverting an interval remove", () => {
+		sharedString.insertText(0, "hello world");
+		const id = collection.add(0, 5, IntervalType.SlideOnRemove).getIntervalId();
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+
+		collection2.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+	});
+	it("remote interval property change interacting with reverting an interval remove", () => {
+		sharedString.insertText(0, "hello world");
+		const id = collection.add(0, 5, IntervalType.SlideOnRemove, { foo: "one" }).getIntervalId();
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+
+		collection2.changeProperties(id, { foo: "two" });
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+		const int = collection.findOverlappingIntervals(0, 5);
```

Comment:
This test fails if I use getIntervalById here instead of findOverlappingIntervals to get the interval. Is that a problem? I expected the new test I added on line 180 to have the same issue, but it doesn't, and I'm confused as to why.

Reply:
What id are you using? At this point `id` is no longer valid since the interval has been recreated with a new id in the revert. The test on line 180 doesn't call getIntervalId; does it fail if you add that call? It should.

You can use the iterator on `collection` instead of findOverlappingIntervals.

Reply:
I was using the id I got back from calling getIntervalId on the initially created interval. The test on 180 doesn't fail if I call getIntervalId, or getIntervalById using the old id value. I'll use the collection iterator in both cases though, just to be safe.

Reply:
An interval with the old id should never be found in the collection after it's removed. Can you debug that and see what's going on?

Reply:
It looks like the interval isn't found, but we don't fail when removeIntervalById returns undefined. There's also already an entry in revertibles, since only the remove edit is undone the first time, so the test just undoes the change that's in revertibles and doesn't fail.